client/state/plugins/installed/selectors.js

```diff
@@ -129,16 +129,18 @@ export function getPluginsWithUpdates( state, siteIds ) {
 }
 
 export function getPluginsOnSites( state, plugins ) {
-	return Object.values( plugins ).reduce( ( acc, plugin ) => {
-		const siteIds = Object.keys( plugin.sites );
-		acc[ plugin.slug ] = getPluginOnSites( state, siteIds, plugin.slug );
-		return acc;
-	}, {} );
```

Comment:
Consulted chatgpt about this function when sites are large and the reduce function is not the most performant. Here's a suggested more performant implementation

```js
export function getPluginsOnSites(state, plugins) {
    const pluginArray = Object.values(plugins);

    // Optionally pre-cache the siteIds if they won't change
    pluginArray.forEach(plugin => {
        if (!plugin.cachedSiteIds) {
            plugin.cachedSiteIds = Object.keys(plugin.sites);
        }
    });

    // Use map instead of reduce to avoid side-effects
    const result = {};
    pluginArray.forEach(plugin => {
        result[plugin.slug] = getPluginOnSites(state, plugin.cachedSiteIds, plugin.slug);
    });

    return result;
}
```

Reply:
Interesting, did it work as expected?